crates/dekaf/src/read.rs

```diff
@@ -59,10 +65,71 @@ impl Read {
             uuid_ptr: collection.uuid_ptr.clone(),
             value_schema: collection.value_schema.clone(),
             value_schema_id,
+
+            journal_name: partition.spec.name.clone(),
+            client,
+        }
+    }
+
+    async fn skip_to_next_doc(&mut self) -> anyhow::Result<()> {
```

Comment:
In [this commit](https://github.com/estuary/flow/pull/1557/commits/1a971b6775793e3304e864f95bb648daa7153e95), I update `gazette` to:

> Refactor streaming error handling within the crate, to remove back-off
timers and policy around which errors are terminal or may be retried.
Instead, surface all errors in the response Stream, and let the client
decide to retry by polling the Stream again, or cancel by dropping the
Stream. Add Error::is_transient() to simplify client checks of common
retry-able error conditions.

I'm looking at this handling, and thinking that this pattern should be extended all the way down into JSON document transcoding: if a JSON parse error weren't terminal for a ReadJsonLines stream, then you could simply handle / log the error and decide that _in this context_ you'd like to poll it again to get the document at the newline _after_ the current (invalid) document.

Reply:
This is implemented here: https://github.com/estuary/flow/pull/1589

Want to pull it in and give it a try?

Reply:
Done!

```
2024-08-29T21:11:38.494853Z  WARN serve{addr=[::ffff:127.0.0.1]:51954}:handle_api{api_key=FetchKey v=11}: dekaf::session: discarding pending read due to offset jump topic="demo.2Fwikipedia.2Frecentchange.2Dsampled" partition=0 old_offset=14064267732 new_offset=14064264074
2024-08-29T21:11:38.578116Z DEBUG next_batch{journal_name="demo/wikipedia/recentchange-sampled/pivot=00"}: dekaf::read: Ignoring first parse error to skip past partial document err=failed to parse document at journal offset range 14064264074..14064264125
2024-08-29T21:11:38.578517Z DEBUG next_batch{journal_name="demo/wikipedia/recentchange-sampled/pivot=00"}: dekaf::read: returning records count=3 first_offset=14064265337 last_offset=14064267731 last_write_head=14064267801 ratio=0.4885906040268456 records_bytes=2234
```